File: node.js/listener.js
const express = require('express');
const fs = require('fs');
const nodemailer = require('nodemailer');
const app = express();
const TOKEN_CORRETO = "TOKEN_DO_CLIENTE_EXEMPLO_ABC123"; // O token individual desse cliente

app.use(express.json());

app.post('/', async (req, res) => {
    const token = req.headers['x-webhook-token'];
    if (token !== TOKEN_CORRETO) {
        return res.status(403).send('Acesso negado!');
    }
    // Grava log
    const log = `---- ${new Date().toISOString()} ----\nRAW: ${JSON.stringify(req.body)}\n------------------------\n`;
    fs.appendFileSync('webhook_log.txt', log);

    // Envia email
    let transporter = nodemailer.createTransport({ host: "localhost", port: 25, secure: false });
    let mailOptions = {
        from: '"Webhook" <webhook@seudominio.com>',
        to: "alvi@alaj.com.br",
        subject: "Webhook recebido!",
        text: "Recebido:\n\n" + JSON.stringify(req.body, null, 2)
    };
    await transporter.sendMail(mailOptions);

    res.send('OK');
});

app.listen(8080, () => console.log('Webhook listener na porta 8080'));

File: roby/listener.rb
require 'sinatra'
require 'mail'
TOKEN_CORRETO = "TOKEN_DO_CLIENTE_EXEMPLO_ABC123" # O token individual desse cliente

Mail.defaults { delivery_method :smtp, address: "localhost", port: 25 }

post '/' do
  token = request.env["HTTP_X_WEBHOOK_TOKEN"]
  halt 403, "Acesso negado!" unless token == TOKEN_CORRETO
  data = request.body.read
  File.open('webhook_log.txt', 'a') do |f|
    f.puts "---- #{Time.now} ----\nRAW: #{data}\n------------------------\n"
  end
  Mail.deliver do
    from     'webhook@seudominio.com'
    to       'alvi@alaj.com.br'
    subject  'Webhook recebido!'
    body     "Recebido:\n\n#{data}"
  end
  "OK"
end
